pkg/plugins/builtin/email/email.go

```diff
@@ -82,42 +86,32 @@ func exec(stepName string, config interface{}, ctx interface{}) (interface{}, in
 
 	message := mail.NewMessage()
 
-	recipients := make([]string, len(cfg.To))
-	for i, recipient := range cfg.To {
-		recipients[i] = message.FormatAddress(recipient, "")
-	}
-
 	message.SetAddressHeader("From", cfg.FromAddress, cfg.FromName)
-	message.SetHeader("To", recipients...)
+	message.SetHeader("To", cfg.To...)
 	message.SetHeader("Subject", cfg.Subject)
 	message.SetBody(
 		http.DetectContentType([]byte(cfg.Body)),
 		cfg.Body,
 	)
 
+	// port and skipTLS already checked at validConfig() lvl
+	// values must be correct so errors are not evaluated
 	port, _ := strconv.ParseUint(cfg.SMTPPort, 10, 64)
 	skipTLS, _ := strconv.ParseBool(cfg.SMTPSkipTLSVerify)
 
 	d := mail.NewDialer(cfg.SMTPHostname, int(port), cfg.SMTPUsername, cfg.SMTPPassword)
-	d.TLSConfig = &tls.Config{InsecureSkipVerify: skipTLS}
+	d.TLSConfig = &tls.Config{InsecureSkipVerify: skipTLS, ServerName: cfg.SMTPHostname}
 	if err := d.DialAndSend(message); err != nil {
-		fmt.Errorf("Send email failed: %s", err.Error())
+		return nil, nil, fmt.Errorf("Send email failed: %s", err.Error())
 	}
 
-	// to reuse configuration
-	parameters := struct {
-		FromAddress string   `json:"from_address"`
-		FromName    string   `json:"from_name"`
-		To          []string `json:"to"`
-		Subject     string   `json:"subject"`
-		Body        string   `json:"body"`
-	}{
+	params := &mailParameters{
 		cfg.FromAddress,
 		cfg.FromName,
 		cfg.To,
 		cfg.Subject,
 		cfg.Body,
 	}
 
-	return &parameters, nil, nil
+	return params, nil, nil
```

Comment:
I believe you can just return cfg.mailParameters :)

Reply:
my bad, thanks.